pkg/azuredisk/azure_controller_common.go

```diff
@@ -250,14 +247,20 @@ func (c *controllerCommon) AttachDisk(ctx context.Context, diskName, diskURI str
 		}
 	}()
 
-	var future *azure.Future
-	future, err = vmset.AttachDisk(ctx, nodeName, diskMap)
-	if err != nil {
-		return -1, err
-	}
-	// err will be handled by waitForUpdateResult below
-	if err = c.waitForUpdateResult(ctx, vmset, nodeName, future, err); err != nil {
-		return -1, err
+	if derr := kwait.ExponentialBackoffWithContext(ctx, updateVMBackoff, func(ctx context.Context) (bool, error) {
+		err = vmset.AttachDisk(ctx, nodeName, diskMap)
+		if err != nil {
+			if IsOperationPreempted(err) {
+				klog.Errorf("Retry VM Update on node (%s) due to error (%v)", nodeName, err)
+				_ = vmset.DeleteCacheForNode(string(nodeName))
```

Comment:
shoud use UpdateVM only operation since operation is preempted

Reply:
updated. Thanks!

Reply:
pls also remove L255 `vmset.DeleteCacheForNode(string(nodeName))`

Reply:
updated